src/hub/ShareClassManager.sol

```diff
@@ -297,13 +297,14 @@ contract ShareClassManager is Auth, IShareClassManager {
         // NOTE: shares and pool currency have the same decimals - no conversion needed!
         payoutPoolAmount = navPoolPerShare.mulUint128(epochAmounts.approvedShareAmount, MathLib.Rounding.Down);
 
-        payoutAssetAmount = PricingLib.poolToAssetAmount(
-            payoutPoolAmount,
+        payoutAssetAmount = PricingLib.shareToAssetAmount(
+            epochAmounts.approvedShareAmount,
             hubRegistry.decimals(poolId),
             hubRegistry.decimals(payoutAssetId),
             epochAmounts.pricePoolPerAsset,
+            epochAmounts.navPoolPerShare,
             MathLib.Rounding.Down
-        ).toUint128();
+        );
```

Comment:
Same calculation but in 1 step instead of 2 which is more precise. Forgot to update this in last pricing improvements.